java/src/main/java/com/powsybl/dataframe/network/adders/AbstractSimpleAdder.java

```diff
@@ -30,4 +32,8 @@ public void addElements(Network network, List<UpdatingDataframe> dataframes) {
     }
 
     public abstract void addElements(Network network, UpdatingDataframe dataframe);
+
+    public void addElementsWithBay(Network network, UpdatingDataframe dataframe, String busbarSectionId, int injectionPositionOrder,
+                                   ConnectablePosition.Direction direction, boolean throwException, Reporter reporter) {
+    }
```

Comment:
I added this method to make it easier, but actually it doesn't apply to every elements. I don't know how to make it look better...

Reply:
Maybe you can add the attributes busbarSectionId, injectionPositionOrder and direction in the dataframe, what do you think ?

Reply:
In addition of the dataframe stuff, we can probably share more code since the only difference between adding with bays or not is the "addition" mechanism: we could have something like
```java
@FunctionalInterface
interface AdditionStrategy {
    void add(UpdatingDataframe dataframe, InjectionAdder<?> adder);  //adds with or without bay
}

//This is still implemented in subclasses
public abstract void addElements(Network network, UpdatingDataframe dataframe, AdditionStrategy addition);

public void addElementsWithBay(Network network, UpdatingDataframe dataframe) {
    addElements(network, dataframe, // method which adds with bays);
}

public void addElements(Network network, UpdatingDataframe dataframe) {
    addElements(network, dataframe, (dataframe, adder) -> adder.add()); // (well actually there is no .add, you need to do like in CreateFeederBay....)
}
```

Reply:
Actually, the dataframe with the injection attributes are supposed to represent the InjectionAdder. So I think it's better like that. To be discussed is if we want to put the attributes necessary to computation like the position, direction ect. in another dataframe.

Reply:
About the functional interface, actually it's not that easy because of the boolean throwException and the reporter....